cmd/unused-exporter/exporter_test.go

```diff
@@ -0,0 +1,226 @@
+package main
+
+import (
+	"context"
+	"log/slog"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/grafana/unused"
+	"github.com/grafana/unused/aws"
+	"github.com/grafana/unused/azure"
+	"github.com/grafana/unused/gcp"
+)
+
+type MockProvider struct {
+	name string
+}
+
+type MockDisk struct {
+	name      string
+	sizeGB    int
+	createdAt time.Time
+	meta      map[string]string
+}
+
+func (p *MockProvider) Name() string {
+	return p.name
+}
+
+func (p *MockProvider) Delete(context context.Context, d unused.Disk) error {
+	return nil
+}
+
+func (p *MockProvider) ID() string {
+	return ""
+}
+
+func (p *MockProvider) ListUnusedDisks(ctx context.Context) (unused.Disks, error) {
+	return nil, nil
+}
+
+func (p *MockProvider) Meta() unused.Meta {
+	return unused.Meta{}
+}
```

Comment:
GoTrick™: given that we are defining `MockProvider` only for a few of the methods, eg, `Name() string`, you can make this much less verbose by doing the following:
```suggestion
type MockProvider struct {
	unused.Provider
	
	name string
}

func (m MockProvider) Name() string { return m.name }

type MockDisk struct {
	name      string
	sizeGB    int
	createdAt time.Time
	meta      map[string]string
}
```

This works because when you _embed_ an interface into a struct, it defines the signatures (sort of); see [this example](https://go.dev/play/p/nwsPyfzDSSx) for a cleaner explanation.

I suppose the same could be done for `MockDisk`.

Reply:
ohhh greatO! tnx merging